src/extension/background-script/actions/ln/sendPaymentOrPrompt.js

```diff
@@ -0,0 +1,49 @@
+import PubSub from "pubsub-js";
+import { parsePaymentRequest } from "invoices";
+import utils from "../../../../common/lib/utils";
+import state from "../../state";
+import db from "../../db";
+
+const sendPaymentOrPrompt = async (message, sender) => {
+  PubSub.publish(`ln.sendPayment.start`, message);
+
+  const paymentRequest = message.args.paymentRequest;
+  const paymentRequestDetails = parsePaymentRequest({
+    request: paymentRequest,
+  });
+
+  const host = message.origin.host;
+  const allowance = await db.allowances
+    .where("host")
+    .equalsIgnoreCase(host)
+    .first();
+
+  if (
+    allowance &&
+    allowance.remainingBudget >= parseInt(paymentRequestDetails.tokens)
+  ) {
+    return sendPaymentWithAllowance(message, paymentRequestDetails, allowance);
+  } else {
+    return payWithPrompt(message);
+  }
+};
+
+async function sendPaymentWithAllowance(
+  message,
+  paymentRequestDetails,
+  allowance
+) {
+  const connector = state.getState().getConnector();
+  const response = await connector.sendPayment({
+    paymentRequest: message.args.paymentRequest,
+  });
+  utils.publishPaymentNotification(message, paymentRequestDetails, response);
+
+  return response;
+}
+
+async function payWithPrompt(message) {
+  await utils.openPrompt({ ...message, type: "sendPayment" });
```

Comment:
I think this is a problem here because now we change the response of the `sendPaymentOrPrompt` call. `sendPaymentWithAllowance` and `payWithPrompt` have different responses. 

the call is initiated from the website with something like the following:

```js
webln.sendPayment(invoice).then(response => { 
  if (verifyPayment(response)) {
    // do something.
  }
}).catch(e => { 
  // show fallback
});
```

so we have to make sure that either way the `sendPayment` promise resolves with the same object.

which might be a problem? hmm..